sdks/java/core/src/main/java/org/apache/beam/sdk/coders/StandardCoder.java

```diff
@@ -228,4 +232,10 @@ public Object structuralValue(T value) throws Exception {
       }
     }
   }
+
+  @SuppressWarnings("unchecked")
+  @Override
+  public TypeDescriptor<T> getEncodedTypeDescriptor() {
+    return (TypeDescriptor<T>) DEFAULT_TYPE_DESCRIPTOR;
```

Comment:
I had explored this path, but this is awkward then when you have:
`MyCoder extends StandardCoder<T>`.

In this case, by default MyCoder would return T when `getEncodedTypeDescriptor` is called (unless it's instantiated as an anonymous subclass: `new MyCoder<String>() {}`).

This is why I thought Object was more appropriate/meaningful.
I am happy to go back to the approach you describe if you think this is better.